exporter/opentelemetry-exporter-otlp-proto-grpc/tests/test_otlp_metrics_exporter.py

```diff
@@ -362,6 +400,36 @@ def test_no_credentials_error(
         OTLPMetricExporter(insecure=False)
         self.assertTrue(mock_ssl_channel.called)
 
+    @patch.dict(
+        "os.environ",
+        {OTEL_EXPORTER_OTLP_METRICS_HEADERS: " key1=value1,KEY2 = VALUE=2 "},
+    )
+    @patch(
+        "opentelemetry.exporter.otlp.proto.grpc.exporter.ssl_channel_credentials"
+    )
+    @patch("opentelemetry.exporter.otlp.proto.grpc.exporter.secure_channel")
+    # pylint: disable=unused-argument
+    def test_otlp_headers_from_env(self, mock_ssl_channel, mock_secure):
+        exporter = OTLPMetricExporter()
+        # pylint: disable=protected-access
+        self.assertEqual(
+            exporter._headers, (("key1", "value1"), ("key2", "VALUE=2"))
+        )
+        exporter = OTLPMetricExporter(
+            headers=(("key3", "value3"), ("key4", "value4"))
+        )
+        # pylint: disable=protected-access
+        self.assertEqual(
+            exporter._headers, (("key3", "value3"), ("key4", "value4"))
+        )
```

Comment:
When headers are set with both env and constructor params, should they be merged?

Reply:
No, one should override the other. Merging is something that only can be done for certain data structures like dictionaries, and the behavior should be the same as other environment variables that have different values, like boolean. It is always the most specific (the constructor parameters) the one who overrides the least specific (the environment variables).

Reply:
Can you add a test for the override? I am not sure where the overriding is happening.

Reply:
This tests the overriding by using constructor-passed headers and then checking that the headers present in the exporter are those that were passed to the constructor and not the ones defined in the environment variables.